src/game/mario_step.c

```diff
@@ -243,7 +243,7 @@ void stop_and_set_height_to_floor(struct MarioState *m) {
     m->vel[1] = 0.0f;
 
     // Hackersm64: this condition fixes potential downwarps
-    if (m->pos[1] <= m->floorHeight + 160.0f) {
+    if ((m->flags & MARIO_TELEPORTING) || m->pos[1] <= m->floorHeight + 160.0f) {
```

Comment:
So this should be reverted to vanilla. The only instance where this is useful seems to be `act_shocked` and even then, `act_shocked` should just skip calling this function entirely